app/controllers/admin/registries_controller.rb

```diff
@@ -60,8 +79,30 @@ def edit
   # given registry. This might change in the future.
   def update
     @registry = Registry.find(params[:id])
-    @registry.update_attributes(update_params)
-    redirect_to admin_registries_path, notice: "Registry updated successfully!"
+    @registry.assign_attributes(update_params)
+    @can_change_hostname = !Repository.any?
+
+    # Check the reachability of the registry.
+    unless params[:force]
+      msg = @registry.reachable?
+      unless msg.empty?
+        logger.info "\nRegistry not reachable:\n#{@registry.inspect}\n#{msg}\n"
```

Comment:
can we merge this code with the one on the create method in an easy way ? I feel it's basically the same thing...